pre-commit

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -eu
+
+git diff "HEAD@{1}" --name-only | grep dune-project 2>&1 > /dev/null
+CHANGED=$?
+
+if [ $CHANGED ]; then
+    echo
+    echo -e "\e[41m!!! dune-project has changed !!!\033[0m"
+    echo
+    echo "You should apply"
+    echo "make"
+    echo "to ensure all the .opam files are changed accordingly"
+fi
```

Comment:
As it is this check is not enough and particularly annoying. It will often print false alert/warnings, and never really prevent bad commits (sinc it never exits with exit code different from 0).
Waht it should rather do is if the dune-project file has changed, then check whether freshly generated `.opam` files have a diff with the current one, and exit if the diff exists.